### PR TITLE
Change core for Monopole and add test

### PR DESCRIPTION
**Pull request by outscale-nts**

### CODE REVIEW

File: src/brick.c [hunk 1]
```diff
@@ -455,11 +455,11 @@ int pg_brick_link(struct pg_brick *west,
 	}
 	/* check if each sides have places */
 	if (!is_place_available(east, WEST_SIDE)) {
-		*errp = pg_error_new("%s: Side full", east->name);
+		*errp = pg_error_new("%s:WEST side full", east->name);
 		return -1;
 	}
 	if (!is_place_available(west, EAST_SIDE)) {
-		*errp = pg_error_new("%s: Side full", west->name);
+		*errp = pg_error_new("%s:EAST side full", west->name);
```
Comment (outscale-jju) [inline src/brick.c]: Could you keep the " " after ":"  and write "east" instead of "EAST" ?

File: src/brick.c [hunk 2]
```diff
@@ -606,8 +609,8 @@ inline int pg_brick_burst(struct pg_brick *brick, enum pg_side from,
 {
 	if (unlikely(!brick))
 		return 0;
-	/* @side is the opposite side of the direction on which
-	 * we send the packets, so we flip it */
+	/* @from is the opposite side of the direction on which
+	* we send the packets, so we flip it */
```
Comment (outscale-jju) [inline src/brick.c]: keep this space :)

File: src/brick.c [hunk 3]
```diff
@@ -704,14 +707,7 @@ uint64_t pg_brick_pkts_count_get(struct pg_brick *brick, enum pg_side side)
...
 		return 0;
-	switch (brick->type) {
-	case PG_MULTIPOLE:
-	case PG_DIPOLE:
-		return rte_atomic64_read(&brick->sides[side].packet_count);
-	case PG_MONOPOLE:
-		return rte_atomic64_read(&brick->side.packet_count);
-	}
-	return 0;
+	return rte_atomic64_read(&brick->sides[side].packet_count);
```
Comment (outscale-jju) [inline src/brick.c]: This should move in an other commit and other pull request.
If we don't use brick.side, should we remove it from struct pg_brick ?

**Comment by outscale-jju** [comment]
Could you rename "core: change in brick to handle monopole brick" to some thing more explicit like "core: fix unlink for monopole bricks"

**Comment by outscale-jju** [comment]
Cloud you rename "core: Add test to relink 2 Monopole" to "core: test monopole unlink"

File: tests/core/test-core.c [hunk 4]
```diff
@@ -830,6 +831,33 @@ static void test_brick_core_verify_re_link(void)
...
+	g_assert(!error);
+
+	/* We relink them to make sure unlink works */
+	pg_brick_link(west_brick, east_brick, &error);
+	g_assert(!error);
+
+	pg_brick_destroy(west_brick);
+	pg_brick_destroy(east_brick);
+
+
```
Comment (outscale-jju) [inline tests/core/test-core.c]: remove those two (almost) blank lines

File: tests/core/test-core.c [hunk 5]
```diff
@@ -830,6 +831,33 @@ static void test_brick_core_verify_re_link(void)
...
+	/* We unlink them  */
+	pg_brick_unlink(west_brick, &error);
+	g_assert(!error);
+
+	/* We relink them to make sure unlink works */
+	pg_brick_link(west_brick, east_brick, &error);
+	g_assert(!error);
+
+	pg_brick_destroy(west_brick);
+	pg_brick_destroy(east_brick);
```
Comment (outscale-jju) [inline tests/core/test-core.c]: You can even test more than one time by doing:
```
for (int i = 0; i < 10; i++) { 
	/* We link the 2 bricks */
	pg_brick_link(west_brick, east_brick, &error);
	g_assert(!error);

	/* We unlink them  */
	pg_brick_unlink(west_brick, &error);
	g_assert(!error);
}
pg_brick_destroy(west_brick);
pg_brick_destroy(east_brick);
```

File: tests/core/test-core.c [hunk 6]
```diff
@@ -830,6 +831,33 @@ static void test_brick_core_verify_re_link(void)
 	pg_brick_destroy(s);
 }
 
+static void test_brick_verify_re_link_monopole(void)
```
Comment (outscale-jju) [inline tests/core/test-core.c]: You don't need to name it "verify" as a test always verify something.
Maybe you can name it "test_brick_monopole_unlink"

File: src/brick.c (hunk 1 above)
Comment (outscale-mgo) [inline src/brick.c]: or you can call pg_side_to_string

**Pull request closed without merging**